File: micro-transaction/src/domain/model/transaction-response.ts
import { TransactionStatus } from "./transaction-status";
import { TransactionType } from "./transaction-type";

export class TransactionResponse {

    transactionExternalId: string;

    transactionType: TransactionType;

    value: number;

    transactionStatus: TransactionStatus;

    createdAt: string;


    constructor(
        transactionExternalId: string,
        transactionType: TransactionType,
        value: number,
        transactionStatus: TransactionStatus,
        createdAt: string
    ) {
        this.transactionExternalId = transactionExternalId;
        this.transactionType = transactionType;
        this.value = value;
        this.transactionStatus = transactionStatus;
        this.createdAt = createdAt;
    }
}

File: yape-financial/yape-transaction/src/main/java/com/tec/yape/transaction/domain/model/TransactionResponse.java
package com.tec.yape.transaction.domain.model;

import com.tec.yape.transaction.domain.enums.TransactionStatus;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
public class TransactionResponse {

    private UUID id;
    private UUID transactionExternalId;
    private UUID accountExternalIdDebit;
    private UUID accountExternalIdCredit;
    private Integer transferTypeId;
    private BigDecimal value;
    private TransactionStatus transactionStatus;
    private LocalDateTime createdAt;

}